src/Input/IntParam.php

```diff
@@ -33,32 +32,44 @@ public function getQuestion(): Question
     {
         $question = $this->createQuestion();
 
-        $question->setNormalizer(static function ($value) {
-            if (is_numeric($value) && (string) (int) $value === $value) {
-                return (int) $value;
+        $question->setNormalizer(
+            /**
+             * @param mixed $value
+             * @return mixed
+             */
+            static function ($value) {
+                if (is_numeric($value) && (string) (int) $value === $value) {
+                    return (int) $value;
+                }
+
+                return $value;
             }
-
-            return $value;
-        });
-
-        $question->setValidator(function ($value) {
-            if (! is_int($value)) {
-                throw new RuntimeException(sprintf(
-                    'Invalid value: integer expected, %s given',
-                    get_debug_type($value)
-                ));
-            }
-
-            if ($this->min !== null && $value < $this->min) {
-                throw new RuntimeException(sprintf('Invalid value %d; minimum value is %d', $value, $this->min));
+        );
+
+        $question->setValidator(
+            /** @param mixed $value */
+            function ($value): int {
+                Assert::integer($value, sprintf('Invalid value: integer expected, %s given', get_debug_type($value)));
```

Comment:
Should the validator throw at all?

Reply:
Yes - this is how validators in symfony/console are expected to work, and it's how we'd coded it previously; I'm just now doing it with assertions.